mh/pat.py

```diff
@@ -2405,6 +2453,21 @@ def notify_layer_departure(self):
             ntc_data = pati.lp2_string(self.session.capcom_id)
             self.server.layer_broadcast(self.session, PatID4.NtcLayerOut,
                                         ntc_data, 0)
+        if self.session.local_info['circle_id'] is not None:
+            self.notify_circle_leave(session.local_info['circle_id'] + 1, 0)
+
+    def notify_circle_leave(self, circle_index, seq):
+        circle = self.session.get_circle()
+        self.sendNtcCircleLeave(circle, circle_index, seq)
+        if circle.leader == self.session:
+            if circle.departed:
+                new_host = self.session.try_transfer_circle_leadership()
+                if new_host:
+                    self.sendNtcCircleHost(circle, new_host[1], new_host[0], seq)
```

Comment:
Please unpack the values rather than using the tuple directly.
```python
index, player = self.session.try_transfer_circle_leadership()
```

Reply:
It returns only a single value, `None,` rather than a tuple if host transferrence is not applicable. I figured this way would be cleaner than returning a tuppe of two `None`s in that case and then checking either one of them.

Reply:
Agreed, I'll switch it over to the `None, None` return